File: dojo_ruby_exemplo/spec/fizz_buzz_spec.rb
require "fizz_buzz"

describe FizzBuzz do
  
  it "Testa se 1 retorna 1" do
    fizz_buzz = FizzBuzz.new
    retorno = fizz_buzz.say 1
    retorno.should == "1"
  end
  
  it "Testa se 2 retorna 2" do
    fizz_buzz = FizzBuzz.new
    retorno = fizz_buzz.say 2
    retorno.should == "2"
  end
  
  it "Testa se 3 retorna Fizz" do
    fizz_buzz = FizzBuzz.new
    retorno = fizz_buzz.say 3
    retorno.should == "Fizz"
  end
  
  it "Testa se 5 retorna Buzz" do
    fizz_buzz = FizzBuzz.new
    retorno = fizz_buzz.say 5
    retorno.should == "Buzz"
  end
  
  it "Testa se 15 retorna FizzBuzz" do
    fizz_buzz = FizzBuzz.new
    retorno = fizz_buzz.say 15
    retorno.should == "FizzBuzz"
  end
end

File: dojo_js_start/spec/FizzBuzzSpec.js
describe("FizzBuzz", function() {
  var fizzBuzz;

  beforeEach(function() {
    fizzBuzz = new FizzBuzz();
  });

  it("Testa se 1 retorna 1", function() {
    var retorno = fizzBuzz.say(1);
    expect(retorno).toEqual(1);
  });

  it("Testa se 2 retorna 2", function() {
    var retorno = fizzBuzz.say(2);
    expect(retorno).toEqual(2);
  });

  it("Testa se 3 retorna Fizz", function() {
    var retorno = fizzBuzz.say(3);
    expect(retorno).toEqual("Fizz");
  });
  
  it("Testa se 5 retorna Buzz", function() {
    var retorno = fizzBuzz.say(5);
    expect(retorno).toEqual("Buzz");
  });
  
  it("Testa se 15 retorna FizzBuzz", function() {
    var retorno = fizzBuzz.say(15);
    expect(retorno).toEqual("FizzBuzz");
  });

});